speaker.go

```diff
@@ -0,0 +1,60 @@
+// Copyright ©2017 The ev3go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package ev3dev
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+	"reflect"
+)
+
+const (
+	tone = 0x02
+
+	ev_snd = 0x12
+	ev_max = 0x1f
+
+	evBufLen = (ev_max + 7) / 8
+)
+
+// Tone plays a tone at the specified frequency from the ev3 speaker.
+// If freq is zero, playing is stopped.
+func Tone(freq uint32) error {
+	ok, err := hasSound()
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return fmt.Errorf("ev3dev: sound events not available for %q", SoundPath)
+	}
+	ev, err := os.OpenFile(SoundPath, os.O_WRONLY, 0)
+	if err != nil {
+		return fmt.Errorf("ev3dev: failed to open sound event device: %v", err)
+	}
+	defer ev.Close()
+
+	var buf [16]byte
+	binary.LittleEndian.PutUint16(buf[8:10], ev_snd)
+	binary.LittleEndian.PutUint16(buf[10:12], tone)
+	binary.LittleEndian.PutUint32(buf[12:16], freq)
+	_, err = ev.Write(buf[:])
+	return err
+}
+
+func hasSound() (bool, error) {
+	ev, err := os.Open(SoundPath)
+	if err != nil {
+		return false, fmt.Errorf("ev3dev: failed to open sound event device: %v", err)
+	}
+	defer ev.Close()
+
+	var buf [evBufLen]byte
+	err = ioctl(ev.Fd(), eviocgbit(0, ev_max), reflect.ValueOf(buf[:]).Index(0).Addr().Pointer())
+	if err != nil {
+		return false, fmt.Errorf("ev3dev: failed to set ioctl command for sound event device: %v", err)
```

Comment:
This reflects the error that is used in `buttons.go`. Fixing the errors that currently use `fmt.Errorf` is part of #24.